// src/scripts/@sipium/sipiumData/gatesConsts.ts
// + 36
export const gatesConsts = {
  1: {
    foodType: 3,
    center: 4,
    physiology: 1,
    products: 1,
    deficit: 1
  },
  2: {
    foodType: 3,
    center: 4,
    physiology: 2,
    products: 2,
    deficit: 2
  },
  3: {
    foodType: 3,
    center: 6,
    physiology: 3,
    products: 3,
    deficit: 3
  },
  5: {
    foodType: 1,
    center: 6,
    physiology: 20,
    products: 12,
    deficit: 12
  },
  8: {
    foodType: 3,
    center: 3,
    physiology: 4,
    products: 2,
    deficit: 2
  },
  9: {
    foodType: 1,
    center: 6,
    physiology: 13
  },
  10: {
    foodType: 3,
    center: 4,
    physiology: 5
  },
  12: {
    foodType: 3,
    center: 3,
    physiology: 6
  },
  14: {
    foodType: 3,
    center: 6,
    physiology: 7
  },
  18: {
    foodType: 1,
    center: 7,
    physiology: 21
  },
  19: {
    foodType: 1,
    center: 5,
    physiology: 22
  },
  20: {
    foodType: 3,
    center: 3,
    physiology: 8
  },
  22: {
    foodType: 3,
    center: 9,
    physiology: 9
  },
  23: {
    foodType: 3,
    center: 3,
    physiology: 4
  },
  24: {
    foodType: 3,
    center: 2,
    physiology: 10
  },
  25: {
    foodType: 3,
    center: 4,
    physiology: 11
  },
  28: {
    foodType: 3,
    center: 7,
    physiology: 12
  },
  29: {
    foodType: 2,
    center: 6,
    physiology: 9
  },
  30: {
    foodType: 2,
    center: 9,
    physiology: 9
  },
  32: {
    foodType: 2,
    center: 7,
    physiology: 24
  },
  34: {
    foodType: 3,
    center: 6,
    physiology: 13
  },
  36: {
    foodType: 2,
    center: 9,
    physiology: 99
  },
  38: {
    foodType: 3,
    center: 5,
    physiology: 14
  },
  39: {
    foodType: 3,
    center: 5,
    physiology: 14
  },
  41: {
    foodType: 2,
    center: 5,
    physiology: 14
  },
  42: {
    foodType: 2,
    center: 6,
    physiology: 13
  },
  43: {
    foodType: 3,
    center: 2,
    physiology: 15
  },
  48: {
    foodType: 1,
    center: 7,
    physiology: 21
  },
  49: {
    foodType: 1,
    center: 9,
    physiology: 9
  },
  51: {
    foodType: 3,
    center: 8,
    physiology: 16
  },
  52: {
    foodType: 1,
    center: 5,
    physiology: 23
  },
  53: {
    foodType: 2,
    center: 5,
    physiology: 25
  },
  54: {
    foodType: 2,
    center: 5,
    physiology: 26
  },
  55: {
    foodType: 3,
    center: 9,
    physiology: 9
  },
  57: {
    foodType: 3,
    center: 7,
    physiology: 17
  },
  58: {
    foodType: 1,
    center: 5,
    physiology: 23
  },
  60: {
    foodType: 3,
    center: 5,
    physiology: 18
  },
  61: {
    foodType: 3,
    center: 1,
    physiology: 19
  }
}
